editor/gui/scene_tree_editor.cpp

```diff
@@ -399,11 +399,16 @@ void SceneTreeEditor::_add_nodes(Node *p_node, TreeItem *p_parent) {
 		if (!scr.is_null()) {
 			String additional_notes;
 			Color button_color = Color(1, 1, 1);
+			StringName class_name = p_node->get_class_name();
 			// Can't set tooltip after adding button, need to do it before.
 			if (scr->is_tool()) {
 				additional_notes += "\n" + TTR("This script is currently running in the editor.");
 				button_color = get_theme_color(SNAME("accent_color"), EditorStringName(Editor));
 			}
+			if (class_name != scr->get_instance_base_type()) {
```

Comment:
This condition has no effect, the lower one is used instead.
Which means it can be removed, because the icon shows correctly anyway 🙃

Reply:
Actually, this is the special case where `custom_type_script` and `script` differ and are not from the same inheritance path. The icon will be displayed in red to indicate that there's an inheritance mismatch. This can happen when the user changes the inheritance in his script that's already attached or drops an incompatible script on a node.

Reply:
Actually, sorry. It's not checking against `custom_type_script`, but the native node type. Would be nice to have an additional check against `custom_type_script` specifically.

Reply:
tbh assigning mismatched type is a known issue that still isn't resolved and probably shouldn't be *in this PR*. Same may apply to custom type script.

Reply:
That's a good point. I'll remove this type checking section in the next commit.